pkg/utils/utils.go

```diff
@@ -327,15 +327,45 @@ func FindNamespace() (string, error) {
 }
 
 func GetDeployments(m *operatorsv1.MultiClusterHub) []types.NamespacedName {
-	return []types.NamespacedName{
+	nn := []types.NamespacedName{
 		{Name: "multiclusterhub-repo", Namespace: m.Namespace},
 	}
+	if m.Enabled(operatorsv1.Insights) {
+		nn = append(nn, types.NamespacedName{Name: "insights-client", Namespace: m.Namespace})
+		nn = append(nn, types.NamespacedName{Name: "insights-metrics", Namespace: m.Namespace})
+	}
+	return nn
+}
+
+func GetFinalizeDeployments(m *operatorsv1.MultiClusterHub) []string {
+	var nn = []string{}
+	if m.Enabled(operatorsv1.Insights) {
+		nn = append(nn, "insights-client")
+		nn = append(nn, "insights-metrics")
+	}
+	return nn
+}
+
+func GetFinalizeServices(m *operatorsv1.MultiClusterHub) []string {
+	var nn = []string{}
+	if m.Enabled(operatorsv1.Insights) {
+		nn = append(nn, "insights-client")
```

Comment:
I am only using these in the finalizer when the MCH is deleted. This was the case where previously I had the Frankenstein approach of deleting services, deployments etc. using owner references. For some reason, those resource types could not be bulk deleted using the installer labels like the other resource kinds so this is the solution I came up with. Do you think it would be better to just call ensureNoInsights within the finalizer logic and remove the resources that way?

Reply:
Yes if ensureNoInsights does the same cleanup then I see no need for duplicating it